.github/workflows/determine-workflow-runner.yml

```diff
@@ -0,0 +1,65 @@
+name: Determine Workflow Runner group
+
+on:
+  workflow_call:
+    inputs:
+      default_runner:
+        description: The runner type that is used by the calling workflow by default
+        required: true
+        type: string
+      large_runner_group_name:
+        description: The name of the runner group that should be used for large jobs
+        required: false
+        type: string
+        default: 'pl-4-core-large-runner'
+      force_large_runner:
+        description: Whether to force the use of a large runner
+        required: false
+        type: boolean
+        default: false
+    outputs:
+      runner_group:
+        description: The runner all subsequent jobs within the calling workflow should run on
+        value: >-
+          ${{
+            jobs.determine_workflow_runner.outputs.runner_group ||
+            (
+              inputs.force_large_runner &&
+              inputs.large_runner_group_name ||
+              inputs.default_runner
+            )
+          }}
+
+jobs:
+  determine_workflow_runner:
+    runs-on: >-
+      ${{
+        (
+          github.event_name == 'pull_request'
+          && contains(github.event.pull_request.labels.*.name, 'urgent')
+        ) && inputs.large_runner_group_name || 'ubuntu-latest'
+      }}
+
+    outputs:
+      runner_group: ${{ steps.runner_group.outputs.runner_group }}
+
+    steps:
+      - name: Output Runner Group name
+        if: >-
+          ${{
+            github.event_name == 'pull_request'
+            && startsWith(inputs.default_runner, 'ubuntu')
+          }}
+        id: runner_group
+        env:
+          # We are not able to use \d to check numeric values as bash does not allow them (not POSIX compliant)
+          RC_BRANCH_FORMAT_REGEX: v[0-9]+\.[0-9]+\.[0-9]+-rc[0-9]?
```

Comment:
I just confirmed our branch protection rules with @Alex-Preciado . This regex works perfectly for our RC branch formatting.

Mildly unrelated, but the regex used for the RC branch protection rule is quite general (`*-rc*`). Maybe worth updating that as well to something similar to the expression used here? I've had instances in the past there I couldn't push to private branches because their name was consistent with the RC branch protection rule.

Reply:
It's a good idea! The branch protection doesn't exactly allow pure regex, but anything that `fnmatch` takes.. [sample syntax here](https://www.geeksforgeeks.org/fnmatch-unix-filename-pattern-matching-python/)